test/extended/cluster-olm-operator/cluster-olm-operator.go

```diff
@@ -0,0 +1,106 @@
+package operatorsv1
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	g "github.com/onsi/ginkgo/v2"
+	o "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	exutil "github.com/openshift/origin/test/extended/util"
+	e2e "k8s.io/kubernetes/test/e2e/framework"
+)
+
+const (
+	packageName = "elasticsearch-operator"
+	version     = "5.8.13"
+)
+
+var _ = g.Describe("[sig-cluster-olm-operator] Cluster OLM Operator", func() {
+
+	var (
+		baseDir   = exutil.FixturePath("testdata", "olmv1")
+		ceFile    = filepath.Join(baseDir, "install-operator.yaml")
+		newCeFile string
+	)
+
+	defer g.GinkgoRecover()
+	oc := exutil.NewCLIWithoutNamespace("default")
+
+	g.It("should set upgradeable true as baseline", func(ctx g.SpecContext) {
+		// Check for tech preview, if this is not tech preview, bail
+		if !exutil.IsTechPreviewNoUpgrade(ctx, oc.AdminConfigClient()) {
+			g.Skip("Test only runs in tech-preview")
+		}
+
+		msg, err := oc.AsAdmin().WithoutNamespace().Run("get").Args("olm", "cluster", "-o=jsonpath={range .status.conditions[*]}{.type}{' '}{.status}").Output()
+		if err != nil {
+			e2e.Failf("Unable to get co %s status, error:%v", msg, err)
+		}
+		o.Expect(err).NotTo(o.HaveOccurred())
+		o.Expect(msg).To(o.ContainSubstring("Upgradeable True"))
```

Comment:
This is a dependency on the order of the status fields. Are we comfortable with that?